packages/deployer/tasks/deploy.js

```diff
@@ -51,13 +51,13 @@ task(TASK_DEPLOY, 'Deploys all system modules')
 
       await hre.run(SUBTASK_PREPARE_DEPLOYMENT, taskArguments);
       await hre.run(SUBTASK_PRINT_INFO, taskArguments);
-      await hre.run(TASK_COMPILE, { force: false, quiet: true });
+      await hre.run(TASK_COMPILE, { force: true, quiet: true });
```

Comment:
This is the real fix for the PR. Before hadhat wasn't refreshing build info when deleting a contract. This will cause the  build to take a little longer, but is necessary for our use case.